tests/unit/Config.js

```diff
@@ -311,4 +311,208 @@ describe('Config', () => {
         const { config } = require('../../lib/Config');
         assert.strictEqual(config.overlayVersion, 0);
     });
+
+    describe('azureArchiveLocationConstraintAssert', () => {
+        const { azureArchiveLocationConstraintAssert } = require('../../lib/Config');
+
+        it('should return undefined if azureStorageEndpoint is missing', () => {
+            const locationObj = {
+                details: {
+                    azureContainerName: 'mycontainer',
+                }
+            };
+            assert.strictEqual(azureArchiveLocationConstraintAssert(locationObj), undefined);
+        });
+
+        it('should return undefined if azureContainerName is missing', () => {
+            const locationObj = {
+                details: {
+                    azureStorageEndpoint: 'http://test.com/',
+                }
+            };
+            assert.strictEqual(azureArchiveLocationConstraintAssert(locationObj), undefined);
+        });
+
+        it('should return undefined if both azureStorageEndpoint and azureContainerName are missing', () => {
+            const locationObj = {
+                details: {
+                }
+            };
+            assert.strictEqual(azureArchiveLocationConstraintAssert(locationObj), undefined);
+        });
+
+        it('should fail assert if azureStorageEndpoint is not a string', () => {
+            const locationObj = {
+                details: {
+                    azureContainerName: 'mycontainer',
+                    azureStorageEndpoint: true,
+                }
+            };
+            try {
+                azureArchiveLocationConstraintAssert(locationObj);
+            } catch (err) {
+                assert.strictEqual(err.message, 'bad config: true must be a string');
+            }
```

Comment:
You've got the wrong assertion here.
We may check that exception, but actually this is not the important part : above all we care that `azureArchiveLocationConstraintAssert` crashes, i.e. that we don't return from `azureArchiveLocationConstraintAssert()` call...

```suggestion
            try {
                azureArchiveLocationConstraintAssert(locationObj);
                assert.fail('azureArchiveLocationConstraintAssert should have asserted !');
            } catch (err) {
                assert.strictEqual(err.message, 'bad config: true must be a string');
            }
```

Reply:
(same for all followup tests)

Reply:
Alternatively, you may use [assert.throws](https://nodejs.org/api/assert.html#assertthrowsfn-error-message) :

```suggestion
            assert.throws(() => azureArchiveLocationConstraintAssert(locationObj));
```

(possibly with 'bad config: true must be a string' as second argument, if you want to check the error message as well)